stdlib/number.gr

```diff
@@ -574,6 +611,260 @@ provide let tan = (radians: Number) => {
+  if (ix >= 0x3fef3333n) {
+    x = pio2_hi - (2.0W * (s + s * r) - pio2_lo)
+  } else {
+    from WasmI64 use { (&) }
+    /* f+c = sqrt(z) */
+    let f = WasmF64.reinterpretI64(
+      WasmI64.reinterpretF64(s) & 0xFFFFFFFF00000000N
+    )
+    let c = (z - f * f) / (s + f)
+    x = 0.5W * pio2_hi -
+      (2.0W * s * r -
+        (pio2_lo -
+          2.0W * c) -
+        (0.5W * pio2_hi -
+          2.0W * f))
+  }
+  x = WasmF64.copySign(x, origAngle)
+  return WasmI32.toGrain(newFloat64(x)): Number
+}
+
+/**
+ * Computes the inverse cosine of the given angle
+ *
+ * @param angle: A number between -1 and 1, representing the angle's cosine value.
+ * @returns The inverse cosine (angle in radians between `-pi/2` and `pi/2`) of x. If x is less than `-1` or greater than `1`, returns NaN
+ *
+ * @since v0.6.0
+ */
+@unsafe
+provide let acos = angle => {
+  // see: musl/src/math/acos.c and SUN COPYRIGHT NOTICE at top of file
+  let origAngle = Numbers.coerceNumberToWasmF64(angle)
+  let mut x = origAngle
+
+  let pio2_hi = 1.57079632679489655800e+00W /* 0x3FF921FB, 0x54442D18 */
+  let pio2_lo = 6.12323399573676603587e-17W /* 0x3C91A626, 0x33145C07 */
+
+  from WasmI32 use {
+    (>>),
+    (&),
+    (|),
+    (!=),
+    leU as (<),
+    geU as (>=),
+    leU as (<=),
+  }
+  from WasmI64 use { (>>>) }
+  from WasmF64 use { (+), (-), (*), (/) }
+
+  let hx = WasmI32.wrapI64(WasmI64.reinterpretF64(x) >>> 32N)
+  let ix = hx & 0x7fffffffn
+
+  /* |x| >= 1 or nan */
+  if (ix >= 0x3ff00000n) {
+    let lx = WasmI32.wrapI64(WasmI64.reinterpretF64(x))
+    from WasmI32 use { (-) }
+    if (WasmI32.eqz(ix - 0x3ff00000n | lx)) {
+      /* acos(1)=0, acos(-1)=pi */
+      if (hx >> 31n != 0n)
+        return WasmI32.toGrain(newFloat64(2.0W * pio2_hi + 0x1p-120W)): Number
+      else return 0
+    }
+    return WasmI32.toGrain(newFloat64(NaNW)): Number
+  }
+  /* |x| < 0.5 */
+  if (ix < 0x3fe00000n) {
+    /* |x| < 2**-57 */
+    let output =
+      if (ix <= 0x3c600000n) pio2_hi + 0x1p-120W
+      else pio2_hi -
+        (x -
+          (pio2_lo -
+            x * rf(x * x)))
+    return WasmI32.toGrain(newFloat64(output)): Number
+  }
+  /* x < -0.5 */
+  if (hx >> 31n != 0n) {
+    let z = (1.0W + x) * 0.5W
+    let s = WasmF64.sqrt(z)
+    let w = rf(z) * s - pio2_lo
+    return WasmI32.toGrain(newFloat64(2.0W * (pio2_hi - (s + w)))): Number
+  }
+
+  /* x > 0.5 */
+  from WasmI64 use { (&) }
+  let z = (1.0W - x) * 0.5W
+  let s = WasmF64.sqrt(z)
+  let df = WasmF64.reinterpretI64(
+    WasmI64.reinterpretF64(s) & 0xFFFFFFFF00000000N
+  )
+  let c = (z - df * df) / (s + df)
+  let w = rf(z) * s + c
+  return WasmI32.toGrain(newFloat64(2.0W * (df + w))): Number
+}
+
+/**
+ * Computes the inverse tangent of the given angle
+ *
+ * @param angle: A number between -1 and 1, representing the angle's tangent value.
+ * @returns The inverse tangent (angle in radians between `-pi/2` and `pi/2`) of x. If x is less than `-1` or greater than `1`, returns NaN
```

Comment:
This is for all the returns in the file.

What is "x"? We usually say something like "of the given angle."

Also, the "if x is less" sentence needs to be something like "The result will be NaN if the given angle is not between -1 and 1." Or something similar

Reply:
I made that change